### PR TITLE
Ignore regtest in get_fee_rate_from_mempool_space

### PR DESCRIPTION
**Pull request by kpp**
# Description

So that e2e tests are not spammed when RUST_BACKTRACE=1. 

```
WARN bitcoin_da::service: Failed to get fee rate from mempool.space e=Unsupported network for mempool space fee estimation

Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
```

### CODE REVIEW

File: crates/bitcoin-da/src/service.rs [hunk 1]
```diff
@@ -569,7 +569,7 @@ impl BitcoinService {
     pub async fn get_fee_rate_as_sat_vb(&self) -> Result<u64, anyhow::Error> {
         // If network is regtest or signet, mempool space is not available
         let smart_fee = match get_fee_rate_from_mempool_space(self.network).await {
-            Ok(fee_rate) => Some(fee_rate),
+            Ok(fee_rate) => fee_rate,
             Err(e) => {
                 tracing::warn!(?e, "Failed to get fee rate from mempool.space");
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/service.rs]: With this change, it might be more appropriate to log error here maybe? Not exactly sure, wdyt?

**Review by yaziciahmet — APPROVED** [review]
Perfecto!

File: crates/bitcoin-da/src/service.rs (hunk 1 above)
Comment (kpp) [reply crates/bitcoin-da/src/service.rs]: Probably yes, you are right

Comment (kpp) [inline crates/bitcoin-da/src/service.rs]:
```suggestion
                tracing::error!(?e, "Failed to get fee rate from mempool.space");
```

**Review by jfldde — APPROVED** [review]

**Pull request merged**